### PR TITLE
feat(CommuterRailTimeTable): Add track reassignment icon

### PR DESCRIPTION
**Pull request by kotva006**
#### Summary of changes

**Asana Ticket:** [Commuter Rail Abnormal Track Assignment](https://app.asana.com/0/555089885850811/1204179568546647)

Adds an Icon to the commuter rail time table to display if the train is arriving at a different track than expected.
Right now this will just say `Track Change` when there is a track change as the new track name isn't exposed to us yet.

---

#### General checks
* [ ] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [ ] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

#### New UI, or substantial UI changes
* [ ] **Cross-browser compatibility** is less of an issue now that we're no longer supporting IE, but changes still need to work as expected in Safari, Chrome, and Firefox.
* [ ] **Are interactive elements accessible?** This includes at minimum having [relevant keyboard interactions](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#keyboard-interaction) and [visible focus](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus), but can also include verification with screen reader testing.
* [ ] **Other accessibility checks** such as sufficient [color constrast](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#color-contrast), or whether the layout holds up at 200% [zoom level](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus).

#### New endpoints, or non-trivial changes to current endpoints
* [ ] **Have we load-tested** any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [ ] **If this change involves routes,** does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?

### CODE REVIEW

File: apps/site/lib/site_web/controllers/schedule/timetable_controller.ex [hunk 1]
```diff
@@ -58,16 +58,65 @@ defmodule SiteWeb.ScheduleController.TimetableController do
...
+          Route.t(),
+          0 | 1
+        ) :: %{
+          required({Schedules.Trip.id_t(), Stops.Stop.id_t()}) => String.t() | nil
+        }
+  defp track_changes(trip_schedules, route, direction_id) do
+    Map.new(trip_schedules, fn {{trip_id, stop_id}, sch} ->
+      # Fetches all the predictions on the route in a direction, for a trip
+      predictions =
+        Predictions.Repo.all(
```
Comment (kotva006) [inline apps/site/lib/site_web/controllers/schedule/timetable_controller.ex:85]: question: Do we want this to use the non-cached version of predictions?  I am not sure the TTL of the cache is short enough for what this feature wants.

File: apps/site/lib/site_web/controllers/schedule/timetable_controller.ex [hunk 2]
```diff
@@ -58,16 +58,65 @@ defmodule SiteWeb.ScheduleController.TimetableController do
...
+  defp get_track_change(schedule, predicted_schedules) do
+    # Comparing stops based off stop sequence
+    prediction =
+      Enum.find(predicted_schedules, fn ps ->
+        ps.stop_sequence == schedule.stop_sequence
+      end)
+
+    # If the prediction stop doesn't match the scheduled stop there has been a track change
+    if prediction != nil and schedule.stop.id != prediction.stop.id do
+      "Track Change"
```
Comment (kotva006) [inline apps/site/lib/site_web/controllers/schedule/timetable_controller.ex:114]: The thought is this should be replaced with the new Track Name and everything should work

File: apps/site/test/site_web/views/schedule/timetable_view_test.exs [hunk 3]
```diff
@@ -101,5 +102,28 @@ defmodule SiteWeb.Schedule.TimetableViewTest do
...
+        Keyword.merge(assigns,
+          header_schedules: header_schedules,
+          track_changes: track_changes,
+          all_stops: all_stops,
+          trip_schedules: trip_schedules
+        )
+
+      rendered = SiteWeb.ScheduleView.render("_timetable.html", assigns)
+      assert safe_to_string(rendered) =~ "Track Change"
+    end
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/test/site_web/views/schedule/timetable_view_test.exs:127]: here seems most appropriate to me.

**Comment by kotva006** [comment]
Closing as this is no longer relevant

**Pull request closed without merging**